src/fides/api/ops/models/privacy_preference.py

```diff
@@ -0,0 +1,206 @@
+    privacy_modal = "privacy_modal"
+
+
+class PrivacyPreferenceHistory(Base):
+    """The DB ORM model for storing PrivacyPreferenceHistory, used for saving
+    every time consent preferences are saved for reporting purposes.
+
+    add fides_user_provided_identity nullable FK
+    add fides_user_device_id encrypted field
+    add ip address
+    add consent preference method?
+    """
+
+    affected_system_status = Column(
+        MutableDict.as_mutable(JSONB), server_default="{}", default=dict
+    )
+    email = Column(
+        StringEncryptedType(
+            type_in=String(),
+            key=CONFIG.security.app_encryption_key,
+            engine=AesGcmEngine,
+            padding="pkcs5",
+        ),
+        nullable=True,
+    )
+    phone_number = Column(
+        StringEncryptedType(
+            type_in=String(),
+            key=CONFIG.security.app_encryption_key,
+            engine=AesGcmEngine,
+            padding="pkcs5",
+        ),
+        nullable=True,
+    )
+    preference = Column(EnumColumn(UserConsentPreference), nullable=False, index=True)
+    privacy_notice_history_id = Column(
+        String, ForeignKey(PrivacyNoticeHistory.id), nullable=False, index=True
+    )
+    privacy_request_id = Column(
+        String, ForeignKey(PrivacyRequest.id), nullable=True, index=True
+    )
+    provided_identity_id = Column(
+        String,
+        ForeignKey(ProvidedIdentity.id),
+        nullable=True,
+    )
+    relevant_systems = Column(
+        MutableList.as_mutable(ARRAY(String))
+    )  # Systems whose data use match.  This doesn't necessarily mean we propagate. Some may be intentionally skipped later.
+    request_origin = Column(EnumColumn(RequestOrigin))
+    secondary_user_ids = Column(
+        MutableDict.as_mutable(
+            StringEncryptedType(
+                JSONTypeOverride,
+                CONFIG.security.app_encryption_key,
+                AesGcmEngine,
+                "pkcs5",
+            )
+        ),
+        nullable=True,
+    )  # Cache secondary user ids (cookies, etc) if known for reporting purposes.
+
+    url_recorded = Column(String)
+    user_agent = Column(
+        StringEncryptedType(
+            type_in=String(),
+            key=CONFIG.security.app_encryption_key,
+            engine=AesGcmEngine,
+            padding="pkcs5",
+        ),
+        nullable=True,
+    )
+
+    user_geography = Column(EnumColumn(PrivacyNoticeRegion), index=True)
+
+    # Relationships
+    privacy_notice_history = relationship(PrivacyNoticeHistory)
+    privacy_request = relationship(PrivacyRequest, backref="privacy_preferences")
+    provided_identity = relationship(ProvidedIdentity)
+    current_privacy_preference = relationship(  # Only exists if this is the same as the Current Privacy Preference
+        "CurrentPrivacyPreference",
+        back_populates="privacy_preference_history",
+        cascade="all, delete",
+        uselist=False,
+    )
+
+    @classmethod
+    def create(
```

Comment:
What do you think of the decision to collapse CurrentPrivacyPreference records based on notice instead of notice history id? While we're saving privacy preferences generally against the historical record, in my opinion it made sense to make the notice unique here.  

Trying to think of what the UI would need to surface "current preferences".   Say the most current preferences we're displaying in the UI are an Advertising notice v6 but we only have preferences saved against v5.  I'd think the v5 preference is still more relevant than whatever the default preference is in that case.